## **Importing necessary libraries**

In [2]:
# import the important packages
import pandas as pd  # library used for data manipulation and analysis

import numpy as np  # library used for working with arrays

import matplotlib.pyplot as plt  # library for visualization

import seaborn as sns  # library for visualization

%matplotlib inline

import scipy.stats as stats  # this library contains a large number of probability distributions as well as a growing library of statistical functions

# Sampling Distribution

### Q1. Suppose an automobile battery manufacturer claims that the mean lifetime of their battery is 60 months with a standard deviation of 6 months. Suppose the distribution of battery life is approximately normal. Find the probability that the mean lifetime of 40 randomly sampled batteries will be less than 58 months.

In [3]:
# import the required function
from scipy.stats import norm
# declare the value of mean lifetime of battery in mu
mu = 60
# declare the value of standard deviation of battery
sigma = 6
# sample size
n = 40
# find the sample standard deviation
s = sigma/np.sqrt(40)
# find the probability
round(norm.cdf(58, loc = mu, scale = s), 4)

0.0175

Insight:

The probability that the mean lifetime of 40 randomly sampled batteries will be less than 58 months is 1.75%.

# Interval Estimation

### Q2. A random sample of 40 households was selected as part of a study on electricity usage, and the number of kilowatt-hours (kWh) was recorded for each household in the sample for the first quarter of 2020. The average usage was found to be 310 kWh. In a very large study in the first quarter of the previous year, it was found that the standard deviation of the usage was 89 kWh.

### Assuming the standard deviation is unchanged and that the usage is normally distributed, provide an expression for calculating a 95% confidence interval for the mean usage in the first quarter of 2019. 

In [4]:
#import the required function
from scipy.stats import norm
import numpy as np

#set the values of sample mean and sigma
x_bar, sigma = 310, 89

# set the value of sample size
n = 40

# construct the confidence interval
np.round(norm.interval(0.95, loc = x_bar, scale = sigma/np.sqrt(n)), 2)

array([282.42, 337.58])

 **Insight:** 
 
 95% of the time, the mean usage in the first quarter of 2019 will be between 282.42 and 337.58 kWh.

# Hypothesis Testing

### Q3. You are a manager of a Chinese restaurant. You want to determine whether the mean waiting time to place an order has changed in the past month from its previous population mean value of 4.5 minutes. State the null and alternative hypotheses.

The null hypothesis is that the mean waiting time has not changed from its previous value of 4.5 minutes. This is stated 
as

$H_0$: µ = 4.5

The alternative hypothesis is that the mean waiting time has been changed from its previous value of 4.5 minutes. This is stated as

$H_a$: µ $\neq$ 4.5

### Q4. What is the p-value if, in a two-tailed hypothesis test, z-stat = +2.00?

In [5]:
# Calculating the one-tailed p-value for z-stat=2
p_val = 1 - stats.norm.cdf(2)
p_val

0.02275013194817921

In [6]:
#since it is a two tailed test
p_val*2

0.04550026389635842

In a two-tailed test, the p-value for the z-stat 2 is = 0.0455

### Q5. Samy, Product Manager of K2 Jeans, wants to launch a product line into a new market area. A Survey of a random sample of 400 households in that market showed a mean income per household of 30000 rupees. The standard deviation based on an earlier pilot study of households is 8000 rupees. Samy strongly believes the product line will be adequately profitable only in markets where the mean household income is greater than 29000 rupees. Samy wants our help in deciding whether the product line should be introduced in the new market. Perform statistical analysis and based on that draw a conclusion.

Null Hypothesis: Mean income of the household is less than or equal to 29000 

$H_0$: µ <= 29000 

Alternate Hypothesis: Mean income of the household is greater than 29000

$H_a$: µ > 29000 

x̅= 30000

μ = 29000

σ= 8000

n = 400

𝑍=(x̅ − μ)/ (σ/(sqrt(n)) = (30000 – 29000) / (8000/sqrt(400))= 2.5

In [7]:
# Calculating the p-value for Zstat=2.5
1 - stats.norm.cdf(2.5)

0.006209665325776159

Interpretation of p-value: 
If the p-value is less than α(significance level), reject the null hypothesis.

Since α is not specified, we assume it to be 0.05. 

Since p-value (0.006) < α (0.05), we reject the null hypothesis and conclude that the mean income of the household is greater than 29000.